ourjapanlife/findadoc-server
# feat: implement end-to-end testing with Jest and Supertest

Resolves #78
Resolves #226 
Resolves #232 

# What changed
1. Creates a Firestore image on Docker to run an emulator for testing.
2. Moves the credentials and environment variables from database.ts to firebaseConfig.ts
3. Creates a GitHub action to run the tests when PRs are open on branches or merged to main.
4. Adds the src/errors.ts file for handling custom error messages.
5. Removes the old test files and mock data that were geared towards Prisma and Postgres.

# Testing instructions
1. Run `yarn install` to install the dependencies
2. Run `yarn generate` to make sure you have all of the updated types.
3. Run `docker-compose -f docker-compose-test.yml up -d` to run the Firestore emulator in Docker
4. Run `yarn test` to run the test files on Docker.
